core/domain/user_jobs_one_off_test.py

```diff
@@ -196,6 +196,63 @@ def test_user_contributions_get_created_after_running_the_job(self):
             ['exp_id'])
 
 
+class UserAuthModelOneOffJobTests(test_utils.GenericTestBase):
+    """Tests for the one-off UserAuthModel migration job."""
+
+    USER_EMAIL = 'a@example.com'
+    USER_USERNAME = 'a'
+    NEW_USERNAME = 'new_username'
+
+    def _run_one_off_job(self):
+        """Runs the one-off MapReduce job."""
+        job_id = user_jobs_one_off.UserAuthModelOneOffJob.create_new()
+        user_jobs_one_off.UserAuthModelOneOffJob.enqueue(job_id)
+        self.assertEqual(
+            self.count_jobs_in_taskqueue(
+                taskqueue_services.QUEUE_NAME_ONE_OFF_JOBS), 1)
+        self.process_and_flush_pending_tasks()
+
+    def setUp(self):
+        super(UserAuthModelOneOffJobTests, self).setUp()
+        self.signup(self.USER_EMAIL, self.USER_USERNAME)
+        self.user_a_id = self.get_user_id_from_email(self.USER_EMAIL)
+
+    def test_one_off_job_new_gae_id_creates_new_auth_model_entry(self):
+        user_settings_model = user_models.UserSettingsModel.get_by_id(
+            self.user_a_id)
+        new_gae_id = user_settings_model.gae_id
+        self.assertIsNone(user_models.UserAuthModel.get_by_auth_id(
```

Comment:
This test is actually verifying two behaviors:
1. That prior to running the migration job, no auth model exists for the given user ID
2. That after running the migration job, an auth model now exists for the user ID

Reply:
Separated the 2 tests now.